src/components/TextInput/BaseTextInput/index.native.tsx

```diff
@@ -267,140 +293,148 @@ function BaseTextInput(props) {
                                     // Add disabled color theme when field is not editable.
-                                    props.disabled && styles.textInputDisabled,
+                                    inputProps.disabled && styles.textInputDisabled,
                                     styles.pointerEventsAuto,
                                 ]}
                                 multiline={isMultiline}
-                                maxLength={props.maxLength}
+                                maxLength={maxLength}
                                 onFocus={onFocus}
                                 onBlur={onBlur}
                                 onChangeText={setValue}
                                 secureTextEntry={passwordHidden}
-                                onPressOut={props.onPress}
-                                showSoftInputOnFocus={!props.disableKeyboard}
-                                keyboardType={getSecureEntryKeyboardType(props.keyboardType, props.secureTextEntry, passwordHidden)}
-                                inputMode={!props.disableKeyboard ? props.inputMode : CONST.INPUT_MODE.NONE}
-                                value={props.value}
-                                selection={props.selection}
+                                onPressOut={inputProps.onPress}
+                                showSoftInputOnFocus={!disableKeyboard}
+                                keyboardType={getSecureEntryKeyboardType(inputProps.keyboardType, inputProps.secureTextEntry ?? false, passwordHidden ?? false)}
+                                inputMode={!disableKeyboard ? inputProps.inputMode : CONST.INPUT_MODE.NONE}
+                                value={value}
+                                selection={inputProps.selection}
                                 readOnly={isReadOnly}
-                                defaultValue={props.defaultValue}
+                                defaultValue={defaultValue}
                                 // FormSubmit Enter key handler does not have access to direct props.
                                 // `dataset.submitOnEnter` is used to indicate that pressing Enter on this input should call the submit callback.
-                                dataSet={{submitOnEnter: isMultiline && props.submitOnEnter}}
+                                dataSet={{submitOnEnter: isMultiline && submitOnEnter}}
                             />
-                            {props.isLoading && (
+                            {inputProps.isLoading && (
                                 <ActivityIndicator
                                     size="small"
                                     color={theme.iconSuccessFill}
                                     style={[styles.mt4, styles.ml1]}
                                 />
                             )}
-                            {Boolean(props.secureTextEntry) && (
+                            {Boolean(inputProps.secureTextEntry) && (
```

Comment:
Same here, and in other places:
```suggestion
                            {!!inputProps.secureTextEntry && (
```